### PR TITLE
Remove nullary warning of artifact symbols

### LINKED ISSUE #11962: Spurious side-effecting nullary warning with universal trait
## reproduction steps

```scala
$ scala -Xlint
Welcome to Scala 2.13.1 (OpenJDK 64-Bit Server VM, Java 11.0.7).
Type in expressions for evaluation. Or try :help.

scala> trait T extends Any { def f = println() }
                                 ^
       warning: side-effecting nullary methods are discouraged: suggest defining as `def f()` instead
defined trait T

scala> class C(val x: Any) extends AnyVal with T { override def f = super.f }
                                                                          ^
       warning: side-effecting nullary methods are discouraged: suggest defining as `def super$f()` instead
                                                                ^
       warning: side-effecting nullary methods are discouraged: suggest defining as `def f()` instead
defined class C

```

## problem

```scala
suggest defining as `def super$f()` instead
```

## expectation

Expectation was that turning on `-Xlint` for partests might exercise linting in interesting ways.

### PR DESCRIPTION
**Pull request by changvvb**
Fixes https://github.com/scala/bug/issues/11962

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala [hunk 1]
```diff
@@ -1604,7 +1604,8 @@ abstract class RefChecks extends Transform {
         // this may be the implementation of e.g. a generic method being parameterized
         // on Unit, in which case we had better let it slide.
         val isOk = (
-             sym.isGetter
+          sym.isSuperAccessor
+          || sym.isGetter
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]:
```
        val isOk = (
             sym.isGetter
          || sym.isDefaultGetter
          || sym.allOverriddenSymbols.exists(over => !(over.tpe.resultType =:= sym.tpe.resultType))
          || sym.isArtifact
        )
```
I did not investigate what else is caught by `isArtifact`. Putting the rare case last in short-circuiting boolean expression is always a good idea; though it is the cheap test here. Also keep existing code format style, which is idiosyncratic here.

File: test/files/neg/t11962.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// scalac: -Xlint -Xfatal-warnings
```
Comment (som-snytt) [inline test/files/neg/t11962.scala]:
```
// scalac: -Xlint -Werror -Ystop-after:refchecks
```
Prefer current -Werror. Stopping after refchecks is a nicety only.

**Review by som-snytt — CHANGES_REQUESTED** [review]
Thanks. A couple of tweaks to suggest.

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala (hunk 1 above)
Comment (changvvb) [reply src/compiler/scala/tools/nsc/typechecker/RefChecks.scala]: Yeah, `isArtifact` seems more suitable.

File: test/files/neg/t11962.scala (hunk 2 above)
Comment (changvvb) [reply test/files/neg/t11962.scala]: I followed [CONTRIBUTING.md](https://github.com/scala/scala/blob/2.13.x/CONTRIBUTING.md)  to use -Xfatal-warnings.
> To test that warnings are correctly emitted, use -Xfatal-warnings with a neg test and .check file. The usual way to create a .check file is partest --update-check.

Maybe CONTRIBUTING.md should be updated.

Comment (som-snytt) [reply test/files/neg/t11962.scala]: Thanks for noticing that.

**Review by som-snytt — APPROVED** [review]

**Comment by som-snytt** [comment]
I forgot to ask you to squash to a single commit. Then `push -f` to rewrite history.

**Review by dwijnand — APPROVED** [review]
Thanks!

**Pull request merged**